src/DocBlock/Annotation.php

```diff
@@ -157,7 +157,7 @@ public function getTypes()
      */
     public function setTypes(array $types)
     {
-        $pattern = '/'.preg_quote($this->getTypesContent()).'/';
+        $pattern = '/'.preg_quote($this->getTypesContent(), '/').'/';
```

Comment:
I checked the logic behind, you are right here - under the hood regex filters alfa-numeric.

For the sake of clear code, I'd suggest keeping this. Please confirm if you'd prefer reverting preg_quote changes.